packages/core/upload/admin/src/hooks/useFolders.js

```diff
@@ -0,0 +1,48 @@
+import { useEffect } from 'react';
+import { useQuery } from 'react-query';
+import { useNotifyAT } from '@strapi/design-system/LiveRegions';
+import { useNotification, useQueryParams } from '@strapi/helper-plugin';
+import { useIntl } from 'react-intl';
+import { axiosInstance, getRequestUrl } from '../utils';
+
+export const useFolders = ({ enabled = true }) => {
+  const { formatMessage } = useIntl();
+  const toggleNotification = useNotification();
+  const { notifyStatus } = useNotifyAT();
+  const [{ rawQuery }] = useQueryParams();
+  const dataRequestURL = getRequestUrl('folders');
+
+  const fetchFolders = async () => {
```

Comment:
Can you extract this function into the `./utils/api` folder?

Also what do you think about the following implementation in order to remove the `useEffect` hooks?

```js
const fetchFolders = async (notify, toggleNotification) => {
 try {
  const { data } = await axiosInstance.get(`${dataRequestURL}${rawQuery}`);
 
  notify( ... );
  
 return data
 } catch(err) {
  toggleNotification({ ... });
  throw new Error(err);
}
}

Reply:
I was thinking the same about the `useEffect` hooks but wasn't sure if they have any value. Since it's the same implementation as `useAssets`: do you think we should do it for both?

Reply:
According to the code they don't have any, I think removing them will prevent us from having to test them because they will basically be wrappers of the useQuery hook.

WDYT?

Reply:
I removed the usage of `useEffect` in b6a28c96b4d11e143da38b5c23d0ce1d87cfed96 and I like it 👍🏼 . After looking at it I don't think it's worth extracting it into `utils/` - what would be the value you see in this?

Reply:
I'd say for consistencies reasons this is how it's setup in the other files, at leat I would put the fetchFolders on the outer scope of the hook WDYT?

Reply:
I think that's not possible, since it consumes other hooks such as `useNotifyAT()` or `useNotification()`. We could make these callbacks or react to the state of the promise, but my feeling is that we over-complicate things in terms of readability.